pkg/snmp/traps/config.go

```diff
@@ -60,12 +82,90 @@ func (c *Config) Addr() string {
 	return fmt.Sprintf("%s:%d", c.BindHost, c.Port)
 }
 
-// BuildV2Params returns a valid GoSNMP SNMPv2 params structure from configuration.
-func (c *Config) BuildV2Params() *gosnmp.GoSNMP {
-	return &gosnmp.GoSNMP{
-		Port:      c.Port,
-		Transport: "udp",
-		Version:   gosnmp.Version2c,
-		Logger:    gosnmp.NewLogger(&trapLogger{}),
+func (c *Config) getAgentHostname() string {
+	hostname, err := util.GetHostname(context.TODO())
+	if err != nil {
+		// this scenario is not likely to happen since
+		// the agent cannot start without a hostname
+		hostname = "unknown-datadog-agent"
 	}
+	return hostname
+}
+
+// BuildAuthoritativeEngineID creates a engineID byte-array from the OS hostname
+func (c *Config) BuildAuthoritativeEngineID() [28]byte {
+	engineID := [28]byte{}
+	// First byte is always 0x80
+	// Next four bytes are the Private Enterprise Number (set to an invalid value here)
+	copy(engineID[:5], []byte{0x80, 0xff, 0xff, 0xff, 0xff})
+	copy(engineID[5:], []byte(c.getAgentHostname()))
```

Comment:
What happen if the hostname is longer than 23 chars (28-5) ?

Reply:
`copy` will strip and only take what ever fits in `engineID[5:]`. See tests cases

Reply:
Would it be better to create a hash from hostname, instead of using the hostname directly ? to avoid cases like very long hostnames.

Example if customer have 2 hostnames like:
```
host123456901234569012345690-1
host123456901234569012345690-2
```
They will have the same engineID.

Reply:
Yes that's a good idea

Reply:
Addressed